src/bootstrap/add.js

```diff
@@ -1,32 +1,28 @@
 'use strict'
 
-const promisify = require('promisify-es6')
+const Multiaddr = require('multiaddr')
+const configure = require('../lib/configure')
 
-module.exports = (send) => {
-  return promisify((args, opts, callback) => {
-    if (typeof opts === 'function' &&
-      !callback) {
-      callback = opts
-      opts = {}
+module.exports = configure(({ ky }) => {
+  return async (addr, options) => {
+    if (addr && typeof addr === 'object' && !Multiaddr.isMultiaddr(addr)) {
```

Comment:
Are we trying to support an `options` object as the first argument?  If so, is this necessary?  The spec says [`addr` is not optional](https://github.com/ipfs/interface-js-ipfs-core/blob/master/SPEC/BOOTSTRAP.md#bootstrapadd).

Reply:
It's what the current code does. So you can pass `add(null, { default: true })` or `add({ default: true })` to add the default bootstrappers. Same sort of situation for `rm` (with `all` option).

So `addr` is required _unless_ you're passing the `default` option.

Reply:
https://github.com/ipfs/interface-js-ipfs-core/pull/557

Reply:
IMO we should leave it like this in this PR.

But i agree with alex and we should make an issue to improve this in the future cause from https://github.com/ipfs/interface-js-ipfs-core/blob/master/SPEC/BOOTSTRAP.md#ipfsbootstrapaddaddr-options i would understand i could do `add({ default: true })` maybe we can simplify to: 
if first param is undefined use defaults.

Reply:
Ah, yes, quite right.  Sigh.

Reply:
> But i agree with alex and we should make an issue to improve this in the future

❤️ me too. I'm just trying to get everything switched to async/await first and then take a second pass at API changes.